include/boost/json/conversion.hpp

```diff
@@ -0,0 +1,293 @@
+    convertible to @ref string_view. Otherwise, `value` is equal to `false`.
+    <br>
+
+    Users can specialize the trait for their own types if they don't want them
+    to be treated like strings. For example:
+
+    @code
+    namespace boost {
+    namespace json {
+
+    template <>
+    struct is_string_like<your::string> : std::false_type
+    { };
+
+    } // namespace boost
+    } // namespace json
+    @endcode
+
+    @par Types satisfying the trait
+
+    @ref string,
+    @ref string_view,
+    <a href="https://en.cppreference.com/w/cpp/string/basic_string"><tt>std::string</tt></a>,
+    <a href="https://en.cppreference.com/w/cpp/string/basic_string_view"><tt>std::string_view</tt></a>.
+
+    @see @ref value_from, @ref value_to
+*/
+template<class T, class Enable = void>
+struct is_string_like
+#ifndef BOOST_JSON_DOCS
+    : std::is_convertible<T, string_view>
+{ };
+#else
+;
+#endif
+
+/** Determine if `T` can be treated like a sequence during conversions.
+
+    Given `t`, a glvalue of type `T`, if
+
+    @li given `It`, the type denoted by `decltype(std::begin(t))`,
+        <tt>std::iterator_traits<It>::iterator_category</tt> is well-formed and
+        denotes a type; and
+
+    @li `decltype(std::end(t))` also denotes the type `It`;
+
+    then the trait provides the member constant `value` that is equal to
+    `true`. Otherwise, `value` is equal to `false`.<br>
+
+    Users can specialize the trait for their own types if they don't want them
+    to be treated like sequences. For example:
+
+    @code
+    namespace boost {
+    namespace json {
+
+    template <>
+    struct is_sequence_like<your::container> : std::false_type
+    { };
+
+    } // namespace boost
+    } // namespace json
+    @endcode
+
+
+    @par Types satisfying the trait
+
+    Any <a href="https://en.cppreference.com/w/cpp/named_req/SequenceContainer"><em>SequenceContainer</em></a>,
+    array types.
+
+    @see @ref value_from, @ref value_to
+*/
+template<class T, class Enable = void>
+struct is_sequence_like
+#ifndef BOOST_JSON_DOCS
+    : mp11::mp_all<
+        mp11::mp_valid_and_true<detail::are_begin_and_end_same, T>,
+        mp11::mp_valid<detail::begin_iterator_category, T>>
+{ };
+#else
+;
+#endif
+
+/** Determine if `T` can be treated like a 1-to-1 mapping during
+    conversions.
+
+    Given `t`, a glvalue of type `T`, if
+
+    @li <tt>is_sequence_like<T>::value</tt> is `true`; and
+
+    @li given types `K` and `M`,
+        `std::remove_cvref_t<decltype(*std::begin(t))>` denotes type
+        `std::pair<K, M>`; and
+
+    @li <tt>std::is_string_like<K>::value</tt> is `true`; and
+
+    @li given `v`, a glvalue of type `V`, and `E`, the type denoted by
+        `decltype(t.emplace(v))`,
+        <tt>std::is_tuple_like<E>::value</tt> is `true`;
+
+    then the trait provides the member constant `value`
+    that is equal to `true`. Otherwise, `value` is equal to `false`.<br>
```

Comment:
The alternatives should be "Then this is an alias for `std::true_type`,...  otherwise it is an alias for `std::false_type`"

Reply:
They aren't aliases, they are separate types. I copied the approach to documenting these from cppreference.